src/tools/nbench/benchmark.cpp

```diff
@@ -253,3 +254,125 @@ vector<runtime::PerformanceCounter> run_benchmark(shared_ptr<Function> f,
     vector<runtime::PerformanceCounter> perf_data = backend->get_performance_data(f);
     return perf_data;
 }
+
+void run_benchmark_validation(shared_ptr<Function> f, const string& backend_name)
+{
+    NodeVector new_results;
+    for (auto n : f->get_ordered_ops())
+    {
+        //dont include op::Results otherwise Function c-tor will complain
+        if (!n->is_output() && !n->is_parameter() && !n->is_constant() &&
+            !(n->get_outputs().size() > 1) && n->get_element_type() == element::f32)
+        {
+            // place conditionals here if you want to only make certain ops an output/result node
+            new_results.push_back(n);
+        }
+    }
+
+    //no need to include original results they are subsumed by new_results
+    auto new_func = make_shared<Function>(new_results, f->get_parameters());
+
+    // // uncomment these lines to serialize the new_func for later use
+    // // I use this for splicing a small graph out of a larger one
+    // string js = serialize(new_func, 4);
+    // std::ofstream outfile;
+    // outfile.open("conv_bprop_filters.json");
+    // outfile << js;
+    // outfile.close();
+    // if (new_func) exit(0);
+
+    Uniform<float> rng(1.0f, 2.0f, 2112);
+    vector<vector<float>> args;
+    for (shared_ptr<op::Parameter> param : new_func->get_parameters())
+    {
+        vector<float> tensor_val(shape_size(param->get_shape()));
+        rng.initialize(tensor_val);
+        args.push_back(tensor_val);
+    }
+
+    auto cpu_func = ngraph::clone_function(*new_func);
+    auto bk_func = ngraph::clone_function(*new_func);
+
+    auto cpu_results = execute(cpu_func, args, "CPU");
```

Comment:
This is only valid if CPU backend is built. You need to make this code conditional. Move code to separate source file and only add it if the CPU is built.